project/src/main/java/com/google/sps/data/PlacesScorer.java

```diff
@@ -0,0 +1,59 @@
+package com.google.sps.data;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.stream.Collectors;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+
+import com.google.maps.model.LatLng;
+
+public class PlacesScorer {
+
+    private static final double RATING_WEIGHT = 0.7;
+    private static final double DURATION_WEIGHT = 0.3;
+    private static final double MAX_RATING = 5;
+    private static final double MAX_DURATION_SECONDS = 40 * 60;
+    private ImmutableMap<Place, Double> durations;
+    private ImmutableList<Place> places;
+
+    /**
+     * Constructor for PlacesScorer, calculates the durations map used for scoring.
+     *
+     * @param placesToScore A list of places we want to calculate their score
+     * @param userLocation The user's physical location used for duration calculations
+     */
+    public PlacesScorer(ImmutableList<Place> placesToScore, LatLng userLocation) {
+        this.places = placesToScore;
+        this.durations = getDurations(userLocation);
+    }
+
+   /**
+   * Returns a map of a place and the score the place gets based on a scoring algorithm.
+   * @return A map between a place to a double representing the place’s score
+   */
+    public ImmutableMap<Place, Double> getScores() {
+        Map<Place, Double> scores = new HashMap<>();
```

Comment:
Why create a `HashMap` and then copy it entirely? 

Use `ImmutableMap.Builder` and then `build` it the `return` statement.

Reply:
No reason, it's the examples I saw online. I changed it to builder.